testsuite/tests/parse-errors/unclosed_simple_expr.compilers.reference

```diff
@@ -1,61 +1,87 @@
-Line 5, characters 0-1, Line 5, characters 5-7:  (3; 2;;
-                                                 ^    ^^
-Syntax error: ')' expected, the highlighted '(' might be unmatched
-Line 2, characters 0-5,
-Line 2, characters 10-12:
+Line 5, characters 5-7,
+Line 5, characters 0-1:
```

Comment:
The order of positions has been reversed here and in all the other tests displaying a location / range.
I assume that's a bug.

Reply:
Interestingly, this is not a bug per-se.

The new behavior is to print the locations *in the same order they appear in the message below* (so first the main location, then the sub-locations in order). The current behavior is to print locations *in reverse order*. This makes little sense in general, but on this particular error (which is in fact the only one using sub-locations currently) it matches the left-to-right/top-to-bottom intuitive ordering of locations. Since on this error, the main location (the token that should be ')' but isn't) appears in the source code after the sub-location (the opening '('), we get a reversed order which looks weird.

What I could do is sort the locations lexicographically, instead of printing them in the order they appear in the message. I think the two make sense in general, we just have to choose. Any opinions?

Reply:
Given the constraint on this first step (not change the current code or results too much), I would think that sorting the locations before printing them textually (and highlighting them) would be slightly nicer than the current output, indeed.

Reply:
You could also not print the location that is printed in the sub-message. But I guess that would remove half of the highlighting?

I might still prefer that to adding any kind of sorting code if you indeed fix the highlighting for sub-messages in a later PR.
If you've done that work already and simply don't want it part of the present PR, you could perhaps setup a branch on your repo for us to have a look.

Reply:
To clear things up, here's a summary of the situation.

On the following code:
```ocaml
let x = (1 +
  2 in ();;
```

## 1. Before this PR
### 1.1 In batch mode
(modulo a missing `\n`)
```
File "/tmp/toto.ml", line 2, characters 4-6:
Error: Syntax error: ')' expected
File "/tmp/toto.ml", line 1, characters 8-9:
Error: This '(' might be unmatched
```
### 1.2 In an ANSI toplevel
![scrot1 2](https://user-images.githubusercontent.com/315125/43838452-aac65aac-9b1b-11e8-9de3-3122e593fa36.png)
-> in ANSI mode, no location is printed and all locations are highlighted
### 1.3 In a dumb toplevel
On 4.07:
```
Characters 8-9:
  let x = (1 +
          ^
Syntax error: ')' expected, the highlighted '(' might be unmatched
```
(the sub-location gets highlighted arbitrarily)

On trunk (modulo some missing `\n`):
```
Line 1, characters 8-9,
Line 2, characters 4-6:
  ........(...
  ....in.....
Syntax error: ')' expected, the highlighted '(' might be unmatched
```
-> highlighting in dumb mode prints the location(s)

## 2. With this PR (currently)
### 2.1 In batch mode
```
File "/tmp/toto.ml", line 2, characters 4-6:
Error: Syntax error: ')' expected
       File "/tmp/toto.ml", line 1, characters 8-9:
       This '(' might be unmatched
```
### 2.2 In an ANSI toplevel
![scrot2 2](https://user-images.githubusercontent.com/315125/43839039-654f2434-9b1d-11e8-89fa-5d73f13e40cb.png)
-> in ANSI mode, all locations are highlighted and the main location is not printed
### 2.3 In a dumb toplevel
```
Line 2, characters 4-6,
Line 1, characters 8-9:
  ........(...
  ....in.....
Error: Syntax error: ')' expected
       Line 1, characters 8-9:
       This '(' might be unmatched
```
-> in dumb mode, highlighting includes printing the locations, and the main location is not printed

Reply:
Here's an alternative propositon for 2.3: split the highlighting part across the locations:
```
Line 2, characters 4-6:
    2 in ();;
      ^^
Error: Syntax error: ')' expected
       Line 1, characters 8-9:
         let x = (1 +
                 ^         
       This '(' might be unmatched
```

Reply:
> -> in dumb mode, highlighting includes printing the locations, and the main location is not printed

Isn't it?

> Here's an alternative propositon for 2.3: split the highlighting part across the locations:

I think that's a clear improvement over the current (and previous) situation.
I vote yes.

Reply:
>Isn't it?

Hum, it is indeed. I guess I meant that it is printed as part of the "dumb" highlighting function, but not re-printed after as part of the error (like in ANSI mode).